DemoServer/Controllers/Demos/TextSearch/HighlightQueryResultsMapReduce/HighlightQueryResultsMapReduceController.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System.Threading.Tasks;
+using DemoCommon.Models;
+using DemoServer.Utils.Cache;
+using DemoServer.Utils.Database;
+using DemoServer.Utils.UserId;
+using Microsoft.AspNetCore.Mvc;
+#region Usings
+using System.Linq;
+using System.Collections.Generic;
+using Raven.Client.Documents;
+using Raven.Client.Documents.Session;
+using Raven.Client.Documents.Indexes;
+using Raven.Client.Documents.Queries.Highlighting;
+#endregion
+
+namespace DemoServer.Controllers.Demos.TextSearch.HighlightQueryResultsMapReduce
+{
+    public class HighlightQueryResultsMapReduceController : DemoCodeController
+    {
+        public HighlightQueryResultsMapReduceController(UserIdContainer userId, UserStoreCache userStoreCache, MediaStoreCache mediaStoreCache,
+            DatabaseSetup databaseSetup) : base(userId, userStoreCache, mediaStoreCache, databaseSetup)
+        {
+        }
+
+        #region Demo
+        #region Step_1
+        public class ArtistsAllSongs : AbstractIndexCreationTask<LastFm, ArtistsAllSongs.IndexEntry>
+        #endregion
+        {
+            #region Step_2
+            public class IndexEntry
+            {
+                public string Artist { get; set; }
+                public string AllSongTitles { get; set; }
+            }
+            #endregion
+           
+            public ArtistsAllSongs()
+            {
+                #region Step_3
+                Map = songs => from song in songs
+                    select new IndexEntry
+                    {
+                        Artist = song.Artist,
+                        AllSongTitles = song.Title
+                    };
+                #endregion
+
+                #region Step_4
+                Reduce = results => from result in results
+                    group result by result.Artist into g
+                    select new IndexEntry
+                    {
+                        Artist = g.Key,
+                        AllSongTitles = string.Join(" ", g.Select(x => x.AllSongTitles))
+                    };
+                #endregion 
+                
+                #region Step_5
+                Store(x => x.Artist, FieldStorage.Yes);
```

Comment:
is there a need to store artist on the index?

Reply:
yes - an exception is thrown if you don't